### PR TITLE
Add more info to conservative transfer error

### PR DESCRIPTION
**Pull request by GiudGiud**
Misc PR responding to slack messages asking for things

### CODE REVIEW

File: framework/src/transfers/MultiAppConservativeTransfer.C [hunk 1]
```diff
@@ -153,23 +153,25 @@ MultiAppConservativeTransfer::initialSetup()
     const auto multi_app = hasFromMultiApp() ? getFromMultiApp() : getToMultiApp();
 
     // Let us check execute_on here. Users need to specify execute_on='transfer' in their input
-    // files for the postprocessors that are used to compute conversative qualities Master app
-    FEProblemBase & master_problem = multi_app->problemBase();
+    // files for the postprocessors that are used to compute conservative qualities Parent app
```
Comment (lindsayad) [inline framework/src/transfers/MultiAppConservativeTransfer.C:156]: This comment doesn't make much sense to me

File: framework/src/transfers/MultiAppConservativeTransfer.C [hunk 2]
```diff
@@ -153,23 +153,25 @@ MultiAppConservativeTransfer::initialSetup()
...
         _current_direction == TO_MULTIAPP ? pps_empty : _to_postprocessors_to_be_preserved;
-    for (auto & master_pp : master_pps)
+    for (auto & pp : parent_app_pps)
     {
-      // Get out all execute_on options for master pp
-      auto & execute_on = master_problem.getUserObjectBase(master_pp).getExecuteOnEnum();
-      // Check if master app has transfer execute_on
+      // Get out all execute_on options for parent app source pp
+      auto & execute_on = parent_problem.getUserObjectBase(pp).getExecuteOnEnum();
+      const auto type = parent_problem.getUserObjectBase(pp).type();
```
Comment (lindsayad) [inline framework/src/transfers/MultiAppConservativeTransfer.C:166]:
```suggestion
      const auto & type = parent_problem.getUserObjectBase(pp).type();
```

File: framework/src/transfers/MultiAppConservativeTransfer.C [hunk 3]
```diff
@@ -187,13 +189,17 @@ MultiAppConservativeTransfer::initialSetup()
       {
         // Get out of all execute_on options for sub pp
         auto & execute_on = sub_problem.getUserObjectBase(sub_pp).getExecuteOnEnum();
+        const auto type = sub_problem.getUserObjectBase(sub_pp).type();
```
Comment (lindsayad) [inline framework/src/transfers/MultiAppConservativeTransfer.C:192]:
```suggestion
        const auto & type = sub_problem.getUserObjectBase(sub_pp).type();
```

File: framework/src/transfers/MultiAppConservativeTransfer.C [hunk 4]
```diff
@@ -278,7 +284,7 @@ MultiAppConservativeTransfer::adjustTransferedSolutionNearestPoint(
   else
     from_adjuster = 0;
 
-  /* Everyone on master side should know this value, and use it to scale the solution */
+  /* Everyone on parent application side should know this value, and use it to scale the solution */
```
Comment (lindsayad) [inline framework/src/transfers/MultiAppConservativeTransfer.C:287]:
```suggestion
  /* Everyone on the parent application side should know this value; use it to scale the solution */
```

File: framework/src/transfers/MultiAppConservativeTransfer.C [hunk 5]
```diff
@@ -394,7 +400,7 @@ MultiAppConservativeTransfer::adjustTransferedSolution(FEProblemBase * from_prob
   else
     from_adjuster = 0;
 
-  /* Everyone on master side should know this value, and use it to scale the solution */
+  /* Everyone on parent side should know this value, and use it to scale the solution */
```
Comment (lindsayad) [inline framework/src/transfers/MultiAppConservativeTransfer.C:403]:
```suggestion
  /* Everyone on the parent side should know this value; use it to scale the solution */
```

File: framework/src/transfers/MultiAppConservativeTransfer.C (hunk 1 above)
Comment (GiudGiud) [inline framework/src/transfers/MultiAppConservativeTransfer.C:156]:
```suggestion
    // files for the postprocessors that are used to compute the quantities to conserve in the Parent app
```

**Review by lindsayad — APPROVED** [review]

**Pull request merged**